gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua

```diff
@@ -136,6 +164,9 @@ function _M:access(context)
 
   ensure_report_timer_on(self, service_id, backend)
 
+  -- The caching policy sets a cache handler in the context
+  self.cache_handler = context.cache_handler
```

Comment:
Ideally, the cache handler should be set in the initializer and passed to `BackendDowntimeCache.new()`. Unfortunately, that's not possible because the exported cache_handler is not available there.

Reply:
Is it necessary to mutate `self` ? I know right now each service will get own copy the policy, but that might not be necessary in the future.
I'd like to keep policies "multi tenant" in a way they could operate with the same configuration on multiple services.

Reply:
Right :+1:
Fixed.